packages/grafana-ui/src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,82 @@
+import { css } from '@emotion/css';
+import { FocusScope } from '@react-aria/focus';
+import React, { useState } from 'react';
+import { usePopperTooltip } from 'react-popper-tooltip';
+import { CSSTransition } from 'react-transition-group';
+
+import { ReactUtils } from '../../utils';
+import { Portal } from '../Portal/Portal';
+import { TooltipPlacement } from '../Tooltip/types';
+
+export interface Props {
+  overlay: React.ReactElement | (() => React.ReactElement);
+  placement?: TooltipPlacement;
+  children: React.ReactElement;
+  /** Defaults to click */
+  trigger?: Array<'click' | 'hover'>;
+}
+
+export const Dropdown = React.memo(({ children, overlay, placement, trigger = ['click'] }: Props) => {
+  const [show, setShow] = useState(false);
+  const { getArrowProps, getTooltipProps, setTooltipRef, setTriggerRef, visible } = usePopperTooltip({
+    visible: show,
+    placement: placement,
+    onVisibleChange: setShow,
+    interactive: true,
+    delayHide: 200,
+    delayShow: 150,
+    offset: [0, 8],
+    trigger: trigger,
```

Comment:
❓ Couple of questions:
-  Do we have a way to dismiss the dropdown? (Hitting the `Esc` Key)

-  For users that are using the keyboard, should we implement displaying the dropdown onFocus when `trigger={['hover']}`
![NoDisplayingWithFocusWithKeyboard](https://user-images.githubusercontent.com/239999/180814216-6e75a21b-df43-4d47-97a8-469cc5e517bf.gif)

Reply:
yes, when using hover trigger it should show on focus as well. Not sure we should use hover menus though but if we do it should also show on hover!